### DISCUSSION TITLE
Syslog shows *missing

### DISCUSSION THREAD

**Discussion by Skullack** [Security-Onion-Solutions/securityonion#9571] [Unsupported Versions]
Hey community.

I am getting over 9 million *missing on syslog in 24 hours on the alerts tab and not sure where this is coming from or even how to stop getting that. 

Can anyone point me in a direction on what to look for that is causing this and possible fix. 

![9EE6A6C7-3791-4238-8AA1-56042EB30081](https://user-images.githubusercontent.com/75991195/212234682-c7130c7e-b851-4e28-b308-a3dbf9acfc9d.jpeg)
![6DEC9F64-8630-439A-AC80-D4CA350AB43C](https://user-images.githubusercontent.com/75991195/212234684-a70766c8-c290-449a-9d2b-94fa0eb78fbf.jpeg)

**Comment by InfosecGoon** [comment]
That Sankey diagram seems to indicate that you've got a whole lot of data coming in via syslog, but it's not being recognized so it doesn't have an event.category assigned to it (hence the *Missing). What are you ingesting via syslog in Security Onion?

If you need to see what's coming in, try this query in Hunt: 

event.dataset: "syslog" | groupby source.ip

**Reply by Skullack** [reply]
I am seeing a lot of data from my pfsense firewall.

**Reply by Skullack** [reply]
See attached![image](https://user-images.githubusercontent.com/75991195/214977671-90b62b32-4b12-4f8e-960a-66cc89b1424c.jpeg)![image](https://user-images.githubusercontent.com/75991195/214977672-e8837846-f9d8-48a1-9f38-a06a89aea08d.jpeg)

**Reply by Skullack** [reply]
You are right no event category assigned to it. How would I go about fixing that ?

**Reply by InfosecGoon** [reply]
That's surprising to me, there's a default parser for PFSense data that should be handling that without any configuration. What versions of PFSense and SO are you running? Have you made any changes or customizations to how PFSense is sending its log data?

**Reply by Skullack** [reply]
I did not here is a screenshot of the config in pfsense. ![image](https://user-images.githubusercontent.com/75991195/215140354-17dfca39-78c3-4489-9d57-93db893fc11a.jpeg)![image](https://user-images.githubusercontent.com/75991195/215140405-ff9fd8cb-28d0-4aa6-8d88-50fc1d1688c5.jpeg)![image](https://user-images.githubusercontent.com/75991195/215140424-0dd59751-e6f7-48fa-90e1-2ffbde7386ff.jpeg)

**Reply by Skullack** [reply]
Any suggestion on how to configure SO to identify certain logs that is in CEF format so that it is ingested correctly ? It just shows Syslog with the message containing all the events. I would like to get these events to be properly displayed.

**Reply by InfosecGoon** [reply]
If you want to ingest logs in CEF format, you can use the CEF module for Filebeat.

Elastic docs: https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-cef.html
SO docs on Filebeat modules: https://docs.securityonion.net/en/2.3/filebeat.html#modules
Youtube video (this is for Netflow, but the steps are similar - you don't have to create the pipeline, though, it should already be there): https://www.youtube.com/watch?v=ew5gtVjAs7g

**Reply by Skullack** [reply]
Ah awesome thank you for the direction. I will look into that tonight and hopefully it works out.

**Reply by Skullack** [reply]
to confirm the config for CEF looks like this.
  filebeat:
    third_party_filebeat:
      modules:
        cef:
          log:
            enabled: true
            var.syslog_host: 0.0.0.0

**Reply by Skullack** [reply]
Would I need to change my port on the CEF logs being send. It is currently sending to SO on port 514. Would I need to change it to 9003 ?

**Reply by Skullack** [reply]
Its finally working as it should thanks to you. Logs are starting to look normal and parsing correctly. Thank you for the help and the direction.

**Reply by An0th3rguy** [reply]
Have you managed to map it to the existing syslog port 514 or you did configuration for a new port (9003). If you could provide more details how you got it work. Thanks

**Comment by Skullack** [comment]
> That Sankey diagram seems to indicate that you've got a whole lot of data coming in via syslog, but it's not being recognized so it doesn't have an event.category assigned to it (hence the *Missing). What are you ingesting via syslog in Security Onion?
> 
> 
> 
> If you need to see what's coming in, try this query in Hunt: 
> 
> 
> 
> event.dataset: "syslog" | groupby source.ip

I will try that query and get back to you but I have pfsense sending Syslog to Security Onion. I have changed what is being sent and will see if the logs change. 
Appreciate your help/ input so far.

**Comment by Skullack** [comment]
Standard port and the CEF is working fine with filebeat now. Sent from my iPhoneOn Feb 20, 2023, at 1:26 PM, An0th3rguy ***@***.***> wrote:﻿
Have you managed to map it to the existing syslog port 514 or you did configuration for a new port (9003). Thanks

**Reply by An0th3rguy** [reply]
> Standard port and the CEF is working fine with filebeat now. Sent from my iPhoneOn Feb 20, 2023, at 1:26 PM, An0th3rguy ***@***.***> wrote:﻿ Have you managed to map it to the existing syslog port 514 or you did configuration for a new port (9003). Thanks —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Could you please describe how you did it? I have exactly same problem spent all night configuring but no luck to get it work. Thank you.

**Reply by An0th3rguy** [reply]
My configuration:

filebeat:
  third_party_filebeat:
    modules:
      cef:
        log:
          enabled: true
          var.syslog_host: 0.0.0.0

firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
        hostgroups:
          syslog:
            portgroups:
              - portgroups.syslog
      INPUT:
        hostgroups:
          syslog:
            portgroups:
              - portgroups.syslog

sudo so-firewall addhostgroup syslog
sudo so-firewall addportgroup syslog
sudo so-firewall includehost syslog 192.168.0.0/16
sudo so-firewall addport syslog udp 514

I don`t know what am I missing. Would you please point me to the right direction? Thank you

**Comment by An0th3rguy** [comment]
Any help with configuration would be appreciated. Still unable to get it
work. Thank you

**Reply by InfosecGoon** [reply]
By default, the CEF module for filebeat listens on port 9003 -- you'd need to add a var.syslog_port: #### directive if you want it to listen on something else.

Module docs: https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-cef.html

If you're still having trouble, I'd suggest starting a new thread.
